### PR TITLE
FS-1009 Standardise configuration

### PR DESCRIPTION
**Pull request by gio-karunakaran**
configured flask app to accept new config loader and added dotenv and python files with default environment variables

### CODE REVIEW

File: config/environments/dev.py [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import redis
 from config.environments.default import Config
 from config.utils import VcapServices
+from fsd_tech import configclass
```
[the remark this answers is not in the document]
Comment (gidsg) [reply config/environments/dev.py:8]: Can you put this in the requirements file else it won't work for anyone else!

File: openapi/components.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+components:
```
Comment (gidsg) [inline openapi/components.yml:1]: This is this file changing in this PR?

File: config/environments/dev.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SentientHousePlant) [reply config/environments/dev.py:8]: 👀 See line 37 of requirements.txt

File: requirements.txt [hunk 3]
```diff
@@ -1,32 +1,113 @@
...
+deepdiff==5.7.0
+Deprecated==1.2.13
+distlib==0.3.4
+dnspython==2.2.1
+dparse==0.5.1
+email-validator==1.2.1
+filelock==3.7.1
+Flask==2.0.2
+Flask-Assets==2.0
+Flask-Compress==1.10.1
```
Comment (gidsg) [inline requirements.txt:32]: Why have all these requirements changed?

**Comment by gidsg** [comment]
PR description mentions dotenv but I don't think thie PR actually does anything with dotenv?

File: openapi/components.yml (hunk 2 above)
Comment (gio-karunakaran) [reply openapi/components.yml:1]: This is a small refactor whereby we're separating the api.yml into 2 files 'components.yml' and 'api.yml' and is a convention being used in other FSD repos, we could split this out into another PR if needed? :)

File: requirements.txt (hunk 3 above)
Comment (gio-karunakaran) [reply requirements.txt:32]: I just ran 'pip freeze > requirements.txt' in my sourced in (virtual environment) terminal, I think @harryyo had entered the dependencies manually?

File: README.md [hunk 4]
```diff
@@ -99,3 +99,9 @@ the following while in your virtual enviroment:
 
 Once the above is done you will have autoformatting and pep8 compliance built
 into your workflow. You will be notified of any pep8 errors during commits.
+
+## Pretty Print Config
+
+To print config as a table when runnin flask app, run the following in your terminal:
```
Comment (SentientHousePlant) [inline README.md:105]:
```suggestion
To print config as a table when running `flask app`, run the following in your terminal:
```

File: app.py [hunk 5]
```diff
@@ -33,20 +56,18 @@ def create_app(testing=False) -> Flask:
...
     flask_app.jinja_env.lstrip_blocks = True
 
-    if testing:
-        flask_app.config.from_object(
-            "config.environments.development.DevelopmentConfig"
+    if (os.environ.get("FLASK_ENV") == "development") | testing:
+        flask_app.config.from_object("config.environments.development.DevelopmentConfig"
         )
+        from config.environments.development import DevelopmentConfig
+        if os.environ.get("PRETTY_PRINT"): DevelopmentConfig.pretty_print()
```
Comment (SentientHousePlant) [inline app.py:63]: 🔧 This conditional is included in the fsd_tech library.

File: README.md (hunk 4 above)
Comment (Adam-W1) [inline README.md:105]: typo on 'runnin'

**Review by SentientHousePlant — CHANGES_REQUESTED** [review]
👍 Tiny changes requested :)

File: README.md (hunk 4 above)
Comment (gio-karunakaran) [reply README.md]: 🤦‍♂️ sorted now!

File: app.py (hunk 5 above)
Comment (gio-karunakaran) [reply app.py]: 💯 conditional removed! :)

File: app.py [hunk 6]
```diff
@@ -33,20 +56,18 @@ def create_app(testing=False) -> Flask:
     flask_app.jinja_env.trim_blocks = True
     flask_app.jinja_env.lstrip_blocks = True
 
-    if testing:
-        flask_app.config.from_object(
-            "config.environments.development.DevelopmentConfig"
+    if (os.environ.get("FLASK_ENV") == "development") | testing:
```
Comment (Adam-W1) [inline app.py:62]: just a Q - where do we set the "FLASK_ENV" env var?

Comment (gidsg) [reply app.py:62]: @Adam-W1 I think either in `.flaskenv` file or overridden in an environment variable

File: utils/definitions.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+"""
+Contains some data we use throughout this project.
+Typically this file just contains the project root.
+"""
+from pathlib import Path
+
+
+def get_project_root() -> Path:
+    return Path(__file__).parent.parent
```
Comment (Adam-W1) [inline utils/definitions.py:9]: is this the kind of thing we want to include in our utils across repos?

**Review by Adam-W1 — APPROVED** [review]
LGTM 👌

**Review by SentientHousePlant — APPROVED** [review]

File: utils/definitions.py (hunk 7 above)
Comment (harryyo) [reply utils/definitions.py:9]: FLASK_ROOT exists in the default config as well currently - we should probably choose one or other. Is there an advantage of having it declared in a separate file to other config?

**Review by harryyo — CHANGES_REQUESTED** [review]
Couple of comments

**Pull request merged**